akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/cookbook/RecipeGlobalRateLimit.java

```diff
@@ -0,0 +1,229 @@
+/**
+ *  Copyright (C) 2015 Typesafe <http://typesafe.com/>
+ */
+package docs.stream.cookbook;
+
+import akka.actor.*;
+import akka.dispatch.Mapper;
+import akka.japi.pf.ReceiveBuilder;
+import akka.pattern.Patterns;
+import akka.stream.ActorFlowMaterializer;
+import akka.stream.FlowMaterializer;
+import akka.stream.javadsl.*;
+import akka.stream.javadsl.japi.Creator;
+import akka.stream.testkit.StreamTestKit;
+import akka.testkit.JavaTestKit;
+import akka.util.Timeout;
+import docs.stream.SilenceSystemOut;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import scala.PartialFunction;
+import scala.concurrent.Future;
+import scala.concurrent.duration.Duration;
+import scala.concurrent.duration.FiniteDuration;
+import scala.runtime.BoxedUnit;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static junit.framework.TestCase.assertTrue;
+
+public class RecipeGlobalRateLimit extends RecipeTest {
+  static ActorSystem system;
+
+  @BeforeClass
+  public static void setup() {
+    system = ActorSystem.create("RecipeGlobalRateLimit");
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    JavaTestKit.shutdownActorSystem(system);
+    system = null;
+  }
+
+  final FlowMaterializer mat = ActorFlowMaterializer.create(system);
+
+  static
+  //#global-limiter-actor
+  public class Limiter extends AbstractActor {
+
+    public static class WantToPass { }
+    public static final WantToPass WANT_TO_PASS = new WantToPass();
```

Comment:
Is WantToPass intended to be a singleton instance? Is it possible to use something more idiomatic, such as:

     public static class WantToPass {
         private WantToPass() {}
         public static final WantToPass INSTANCE = new WantToPass();
     }